### ISSUE TITLE
Big picture stats

### ISSUE THREAD

**Issue by zlavergne** [mapbox/osmcha-frontend#200]
## *Brief Description*
Going one step further than saving filters, there should be stats of those filters. A user would be able to gather a plethora of information at a glance just by showing stats about that filter. There are a ton of useful stats that could be shown. 

## What is the motivation / use case for this feature?

1. Users mapping as a team will be able to monitor the big picture of their efforts as well as doing QA on their team.
2. A user notices trouble users (intentional or not) and wants to keep an eye on them, he/she would be able to know at a glance how much havoc they are causing.
3. During a mapathon or even a HOTOSM event, users could see how they're doing as a group and/or what impact they are having.

## What  is the expected behaviour ?

I think it could be implemented in a few different ways:
1. Adding it as a built-in feature of saved filters. Instead of just showing the filters on the user's page, it would show stats about the changesets the filter would query. 
2. Adding it as a separate section on the user's page (perhaps above the saved filters)

In addition to those I think you could show the same stats when a user applies a new filter (not necessarily saved): After a user applies their filters, you simply show the changeset list and large area where the map will render shows your logo with useful links. That would be a very good spot to be able to show stats for a filter that is applied. Perhaps having a tab to switch between the map and stats so they can be revisited.

## Other Information / context:

I hope I submitted this request properly. If not, please let me know.

**Comment by willemarcel** [comment]
Hi @zlavergne !

Thank you for you input! The osmcha API already provides stats, but we don't have a frontend for it yet.

We have [stats for saved filters (AOI)](https://osmcha.mapbox.com/api-docs/#!/aoi/aoi_stats_list) and we have [another endpoint](https://osmcha.mapbox.com/api-docs/#!/stats/stats_list) that can receive the same parameters used in a changeset query.

**Comment by zlavergne** [comment]
@willemarcel Are you guys working on the frontend? I'd be happy to contribute where I can.

**Comment by willemarcel** [comment]
Nice!! Your contributions are welcome, @zlavergne! I believe we don't have anyone working on this. Does @kepta confirm it?

**Comment by kepta** [comment]
@zlavergne would be more than happy to help you out in cutting a PR for this issue. 

Let me know if you need help setting up the project or any clarification on how a certain piece of code works. 

Also, the documentation is non existent 😓. We have plans on improving it in near future.

**Comment by zlavergne** [comment]
@willemarcel @kepta Thanks!

I forked and made a new branch and it's running smoothly on my local machine. 

Is it up to me on how to implement it (based on the options I listed)? Or do I collaborate with you guys?

**Comment by kepta** [comment]
@zlavergne feel free to send a PR, we can have a discussion after that.

**Comment by zlavergne** [comment]
@kepta I haven't contributed to a project like this before. Did I start it properly by forking and making a branch on my fork? What should my next step be? 

I have finally cleared up my plate so I have time to work on this now.

**Comment by willemarcel** [comment]
> @kepta I haven't contributed to a project like this before. Did I start it properly by forking and making a branch on my fork? What should my next step be?

yes, fork, make a new branch, commit your changes to your github repository and create a Pull Request (after you send the first commit to github, a button will appear on the repository on the browser suggesting you to make a Pull Request).

**Comment by rasagy** [comment]
@zlavergne How is it going? Feel free to use this issue to brainstorm further/get feedback on how you want to implement this.

>After a user applies their filters, you simply show the changeset list and large area where the map will render shows your logo with useful links. That would be a very good spot to be able to show stats for a filter that is applied. Perhaps having a tab to switch between the map and stats so they can be revisited.

We initially left the right panel empty/with placeholder info with exactly this intent, and I find that to be a good workflow: It allows you to see stats for any filter that is applied, and these stats should be easy to generate when the API is anyway returning a list of changesets. This would also not limit this to just saved filters, and give a more holistic perspective to someone who opens a link to a AOI/saved filter by another user.

Happy to do some quick sketches on what this could be if that might help you plan this better.

**Comment by zlavergne** [comment]
@rasagy Some sketches would be awesome! My team (a team of mappers who benefit from this tool) is also brainstorming on the best implementation. 

One thing they've noted is they would like to have some sort of visual representation of the stats too. I was thinking a line graph to show the stats over the time of the filter.

**Comment by tridip1931** [comment]
👋  @rasagy It seems this completely went off my radar. The context is super helpful. I'd like to some research and will soon cut a new ticket to capture future conversations.

**Comment by rasagy** [comment]
Sounds great. I have some early ideas that me and @zlavergne discussed: Will clean them up and post here, and happy to work together on this! 👍

**Comment by zlavergne** [comment]
@rasagy I've been swamped since we talked at SOTM. It's looking like my schedule will open up around the end of the year. I'd love to have the sketches we worked up for when I get going again!
